wit/types.wit

```diff
@@ -111,7 +111,7 @@ interface types {
     scheme: option<scheme>,
     authority: option<string>,
     headers: headers
-  ) -> outgoing-request
+  ) -> result<outgoing-request, error>
```

Comment:
nit: consider adding a line of comment describing the failure cases.

Reply:
You're right, but, realistically, someone with a better, deeper understanding of HTTP probably needs to write this comment and describe what precisely is checked eagerly.  More generally, the current comments are not spec-level in precision and formality--they're mostly just enough to let an implementor see how things are supposed to hang together--so we'll need a full pass over the whole interface to get this to a spec-level of quality at some point in the future.

Reply:
I think it is sufficient to just return a result. The error that is returned will contain the provider specific issues. Some providers (e.g. embedded) may not load the certificate roots to support https for example, while other clients will support that.

LGTM.